Understand and tolerate `tailored=false` requests

rtyley opened the pull request:
Part of the changes aim to tolerate `tailored=false`, like in this url:

https://interactive.guim.co.uk/2016/05/brexit-companion/embed/embed.html?tailored=false&beginner=who_can_vote&intermediate=impact_economy&advanced=can_overrule_vote&default=beginner

...which can be produced by https://github.com/guardian/frontend/pull/13157, which sets `tailored=true` to `tailored=false` for 50% of users.

The `id` value can still be used to set an atomId, but if tailored params are present (ie `default`, `beginner`, `intermediate`, etc) they will override the `id` value, even if `tailored=false` or is not even set.

cc @SiAdcock @joelochlann

rtyley committed:
Understand and tolerate `tailored=false` requests

Part of the changes aim to tolerate `tailored=false`, like in this url:

https://interactive.guim.co.uk/2016/05/brexit-companion/embed/embed.html?tailored=false&beginner=who_can_vote&intermediate=impact_economy&advanced=can_overrule_vote&default=beginner

...which can be produced by https://github.com/guardian/frontend/pull/13157,
which sets `tailored=true` to `tailored=false` for 50% of users.

The `id` value can still be used to set an atomId, but if tailored
params are present (ie `default`, `beginner`, `intermediate`, etc) they
will override the `id` value, even if `tailored=false` or is not even set.

SiAdcock commented on src/js/embed.js:
```diff
@@ -63,7 +63,9 @@ function getQueryParams() {
 window.init = function init(parentEl) {
     const params = getQueryParams();
-    const isTailored = !!params.tailored;
+    const isTailored = (params.tailored == 'true');
```
Can we use `===` here?

`params.tailored === 'true'`

joelochlann commented on src/js/embed.js:
```diff
@@ -63,7 +63,9 @@ function getQueryParams() {
-    const isTailored = !!params.tailored;
+    const isTailored = (params.tailored == 'true');
+    const defaultLevel = params.default || 'intermediate';
+    const defaultAtomId = params[defaultLevel] || params.id
```
missing semicolon

joelochlann commented:
👍  looks good, apart from our pedantic manual linting

rtyley committed:
Fix lint issues

http://stackoverflow.com/questions/359494/does-it-matter-which-equals-operator-vs-i-use-in-javascript-comparisons
